### ISSUE TITLE
Environment variables ignored using FastCGI and IIS

### ISSUE THREAD

**Issue by geographika** [MapServer/MapServer#6289]
As per comments at https://github.com/MapServer/MapServer/pull/6284#issuecomment-816565458 there is an issue reading environment variables when using IIS and FastCGI. There are [several places](https://github.com/MapServer/MapServer/search?l=C&q=getenv) in the MapServer codebase where `getenv` is used to retrieve [Environment Variables](https://mapserver.org/environment_variables.html). 

The (copy of) [FastCGI docs](https://github.com/toshic/libfcgi/blob/f3cef74607d4770c590c52fe9b9dd7c702149cd5/doc/FCGI_Accept.3#L59) state:

>      After accepting a new request, FCGI_Accept assigns new values
>      to the global variables stdin, stdout, stderr, and environ.
>      After FCGI_Accept returns, these variables have the same
>      interpretation as on entry to a CGI program.

In summary before the `FCGI_Accept` call in `mapserv.c`  `getenv` works fine:
https://github.com/MapServer/MapServer/blob/cde09201ad2d469325bc911256513c5b9de78694/mapserv.c#L247

However within the `FCGI_Accept` loop `getenv` searches the *request* environment. This means server variables such as `REQUEST_METHOD` , `CONTENT_TYPE` etc. in `cgiutil.c`  work fine when used with `getenv`, but other environment variables such as `CURL_CA_BUNDLE` return `NULL`.  

https://github.com/MapServer/MapServer/blob/cde09201ad2d469325bc911256513c5b9de78694/cgiutil.c#L142

As this seems to be the FastCGI standard I'm not sure why this issue only seems apply to IIS and not other web servers and OSs. 

There is [an example](https://github.com/toshic/libfcgi/blob/f3cef74607d4770c590c52fe9b9dd7c702149cd5/doc/FCGI_Accept.3#L77) of how to save and copy the original environ but this causes HTTP 500 errors on IIS. 

Looking at other examples of FastCGI - MapCache gets its config file prior to `FCGI_Accept`:

https://github.com/MapServer/mapcache/blob/d8d71730e4f03e0b56d1562e484ae0e35f37f96c/cgi/mapcache.c#L253

QGIS Server notes the following:

```c
#ifdef Q_OS_WIN
  // Initialize font database before fcgi_accept.
  // When using FCGI with IIS, environment variables (QT_QPA_FONTDIR in this case) are lost after fcgi_accept().
  QFontDatabase fontDB;
#endif
```

https://github.com/qgis/QGIS/blob/2d1aa68f0d044f2aced7ebeca8d2fa6b754ac970/src/server/qgis_mapserver.cpp#L533

**Comment by geographika** [comment]
Looking through the issues it does appear this has been an ongoing problem: #5463, #4436

Related to `getenv` and MapScript - #3638

**Comment by sdlime** [comment]
Can we close this?

**Comment by geographika** [comment]
Closed with #6305
